### PR TITLE
Maintenance/diverse fixes (e2e, envs in computational services, refresh page, null issue)

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- this PR aims at re-enabling all the computation tests in the CI
  - some tests were not waiting for the computation pipeline to be completed before running the next test (still not bullet proof, the tests are stillskipped)
  - added a test that checks that the pipeline running state is increasing states correctly
- this PR aims at fixing the issues in e2e regarding the pipeline state
  - The pipeline adjacency list is now updated only if it contains data (this is a temporary fix for the state that gets reset when replacing a pipeline)
  - The pipeline running state is now computed in a more robust way (this should fix the e2e tests) 
- this PR sets the ```SIMCORE_NANO_CPUS_LIMIT``` and ```SIMCORE_MEMORY_BYTES_LIMIT``` in the computational service it starts with the limit applied to the service @GitHK, @mguidon 
- this PR catches the issue where the director-v0/-v2 are not responding when opening a project.
  - In this case the project is closed again and a 503 message is sent back (@odeimaiz this is where the frontend should display the message) (fixes [#osparc-issues569](https://github.com/ITISFoundation/osparc-issues/issues/569))
- this PR now generally sets a retry policy with director-v2 calls instead of only a few calls with python syntax issues.
- this PR fixes the issue where refreshing the page would not re-open the project

### CODE REVIEW

**Review by mguidon — APPROVED** [review]
Thanks

**Comment by KZzizzle** [comment]
Thanks Sylvain, the e2e 🎅

File: services/dask-sidecar/src/simcore_service_dask_sidecar/tasks.py [hunk 1]
```diff
@@ -86,32 +91,50 @@ async def _run_computational_sidecar_async(
...
-        ) as sidecar:
-            output_data = await sidecar.run(command=command)
-        log.debug("completed run of sidecar with result %s", f"{output_data=}")
-        return output_data
+    try:
+        async with monitor_task_abortion(
+            task_name=current_task.get_name(),
+        ):
+            _retry = 0
+            _max_retries = 1
```
Comment (GitHK) [inline services/dask-sidecar/src/simcore_service_dask_sidecar/tasks.py:108]: where might these 2 values be used?

File: services/director-v2/src/simcore_service_director_v2/utils/computations.py [hunk 2]
```diff
@@ -10,48 +10,57 @@
 
 log = logging.getLogger(__file__)
 
+_COMPLETED_STATES = (RunningState.ABORTED, RunningState.FAILED, RunningState.SUCCESS)
+_RUNNING_STATES = (RunningState.STARTED, RunningState.RETRY)
+_TASK_TO_PIPELINE_CONVERSIONS = {
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/utils/computations.py:15]: A bit weirdly formatted.
Quesito: could you use sets as keys, since tuples take into consideration the order of the elements.

File: services/web/server/src/simcore_service_webserver/computation_subscribe.py [hunk 3]
```diff
@@ -233,9 +233,8 @@ async def exchange_consumer(
     consumer_running = False
     for task in consumer_tasks:
         task.cancel()
-    with contextlib.suppress(asyncio.CancelledError):
-        for task in consumer_tasks:
-            await task
+    await asyncio.gather(*consumer_tasks, return_exceptions=True)
+
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/computation_subscribe.py:237]: logged_gather maybe here?

File: services/web/server/src/simcore_service_webserver/director_v2_core.py [hunk 4]
```diff
@@ -96,27 +100,27 @@ async def _request_director_v2(
...
-            #   in plain text and this is considered an error
-            # - `director-v2` and `director-v0` can reply with 204 no content
-            if response.status != expected_status.status_code or (
-                response.status != web.HTTPNoContent and isinstance(payload, str)
-            ):
-                raise DirectorServiceError(response.status, reason=str(payload))
-
-            return payload
+        async for attempt in AsyncRetrying(**DEFAULT_RETRY_POLICY):
+            with attempt:
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/director_v2_core.py:105]: so the request retry is applied to all incoming request

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py [hunk 5]
```diff
@@ -176,19 +179,15 @@ async def start_project_interactive_services(
...
+    results = await logged_gather(*start_service_tasks, reraise=True)
+    log.debug("Services start result %s", results)
+    for entry in results:
         if entry:
-            # if the status is present in the results fo the start_service
+            # if the status is present in the results for the start_service
             # it means that the API call failed
             # also it is enforced that the status is different from 200 OK
-            if "status" not in entry:
-                continue
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:188]: Sorry, but why was this removed?

I would guess that the API correctly returns a "status" key. I'd think you want it to fail fast if there is an issue and not just ignore it.

File: services/web/server/src/simcore_service_webserver/projects/projects_handlers.py [hunk 6]
```diff
@@ -519,6 +520,19 @@ async def open_project(request: web.Request) -> web.Response:
 
     except ProjectNotFoundError as exc:
         raise web.HTTPNotFound(reason=f"Project {project_uuid} not found") from exc
+    except DirectorServiceError as exc:
+        # there was an issue while accessing the director-v2/director-v0
+        # ensure the project is closed again
+        await projects_api.try_close_project_for_user(
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/projects_handlers.py:526]: nice

**Review by GitHK — APPROVED** [review]
Looking good. Please find some questions below before merging.

File: services/dask-sidecar/src/simcore_service_dask_sidecar/tasks.py (hunk 1 above)
Comment (sanderegg) [reply services/dask-sidecar/src/simcore_service_dask_sidecar/tasks.py:108]: this is for now a reminder for the dask-sidecar. since currently there is no retry. so not used.

File: services/director-v2/src/simcore_service_director_v2/utils/computations.py (hunk 2 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/utils/computations.py:15]: yes, but sets are unhashable

File: services/web/server/src/simcore_service_webserver/director_v2_core.py (hunk 4 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/director_v2_core.py:105]: yes and the syntax was anyway wrong before. with unbound variables. But still I don't think that thing is necessary.

Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/director_v2_core.py:105]: I kept it so that whatever the idea was, is kept.

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py (hunk 5 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/projects/projects_api.py:188]: not sure I understand your comment.
the code you currently have in master also does not fail but just logs an error.
I only refactored it only.

**Review by odeimaiz — APPROVED** [review]
Working like a charm 👌
